### ISSUE TITLE
Build failure on Mac AARCH64

### ISSUE THREAD

**Issue by fgaz** [LibreSprite/LibreSprite#308]
##### LibreSprite version

```
1.0
```

##### To Reproduce

Try to build on Mac AARCH64.

```
[230/727] Building CXX object src/gfx/CMakeFiles/gfx-lib.dir/region.cpp.opp.o[Klter.cpp.o.op.o
FAILED: src/gfx/CMakeFiles/gfx-lib.dir/region.cpp.o 
/nix/store/hi81iavwbigd3gssp3mw0l90vap8p0ji-clang-wrapper-11.1.0/bin/clang++ -DHAVE_CONFIG_H -DNDEBUG -DSCRIPT_ENGINE_LUA=1 -DUSE_SDL2_BACKEND -I/tmp/nix-build-libresprite-1.0.drv-0/source/third_party/simpleini -I/nix/store/q6x57qp29q49z866sd5gyw73gybb13bi-SDL2-2.0.14-dev/include/SDL2 -I/nix/store/dhcvg5fxpz0p26a37d1xwjqk6jvwq44x-SDL2_image-2.0.5/include/SDL2 -I/tmp/nix-build-libresprite-1.0.drv-0/source/src/. -I/tmp/nix-build-libresprite-1.0.drv-0/source/src/.. -I/tmp/nix-build-libresprite-1.0.drv-0/source/src/../third_party -I/tmp/nix-build-libresprite-1.0.drv-0/source/src/observable -I/tmp/nix-build-libresprite-1.0.drv-0/source/build/src/base -O2 -std=c++14 -stdlib=libc++ -Wall -Wno-switch -std=gnu++11 -O3 -DNDEBUG -MD -MT src/gfx/CMakeFiles/gfx-lib.dir/region.cpp.o -MF src/gfx/CMakeFiles/gfx-lib.dir/region.cpp.o.d -o src/gfx/CMakeFiles/gfx-lib.dir/region.cpp.o -c /tmp/nix-build-libresprite-1.0.drv-0/source/src/gfx/region.cpp
In file included from /tmp/nix-build-libresprite-1.0.drv-0/source/src/gfx/region.cpp:14:
/tmp/nix-build-libresprite-1.0.drv-0/source/src/./gfx/region.h:52:38: error: no viable overloaded '='
      reference operator*() { m_rect = *m_ptr; return m_rect; }
                              ~~~~~~ ^ ~~~~~~
/tmp/nix-build-libresprite-1.0.drv-0/source/src/./gfx/rect.h:17:7: note: candidate function (the implicit copy assignment operator) not viable: no known conversion from 'gfx::details::Box' (aka 'pixman_box32') to 'const gfx::RectT<int>' for 1st argument
class RectT
      ^
1 error generated.
ninja: build stopped: subcommand failed.
```

Full log: https://logs.nix.ci/?key=nixos/nixpkgs.146509&attempt_id=d4cf1bdb-0210-442b-86e9-64f0569930d2

##### Expected behaviour

Libresprite builds

##### OS/Hardware
- Operative System: Mac
- CPU: M1

##### Additional context  

I got this error on a Mac builder while trying to package LibreSprite for NixOS https://github.com/NixOS/nixpkgs/pull/146509#issuecomment-974100427

**Comment by Zughy** [comment]
Hello, thanks for the report.
Unfortunately no maintainer has a Mac set as a development environment (in fact we haven't released 1.0 for Mac), so the only thing I can do is as "help wanted"

**Comment by nogira** [comment]
I've figured out how to build for M1 mac. There was a bug in `manager.cpp`

https://github.com/LibreSprite/LibreSprite/pull/309
